CoreAutomata/Region.cs

```diff
@@ -23,6 +23,19 @@ public Region(int X, int Y, int W, int H)
             this.H = H;
         }
 
+        /// <summary>
+        /// Restrict another Region to be within the bounds of this Region.
+        /// </summary>
+        public Region Clip(Region Region)
+        {
+            var left = Region.X.Clip(X, R - 1);
```

Comment:
This will not work for Regions with X=0 and W=0, because `Clip` will return `-1`. X will be greater than the max value of `-1`.

This situation shouldn't happen, but we could add some Contract.Requires, for example in the setters of W and H. It would be like `Contract.Require(W > 0, "Region widths must be positive");`.

Reply:
I'm leaving it that way for now. And, from my experience, I'd rather avoid `Contract.Requires`, it needs to be activated or something to actually work.